core/src/main/java/org/apache/calcite/rel/core/JoinRelType.java

```diff
@@ -167,19 +168,28 @@ public CorrelateJoinType toLinq4j() {
         "Unable to convert " + this + " to CorrelateJoinType");
   }
 
-  public boolean projectsRight() {
+  /** Transform this JoinRelType to Linq4j JoinType. **/
+  public JoinType toLinq4jJoinType() {
```

Comment:
Similar comment here. How about moving the code inside EnumerableNestedLoopJoin since  it is used only there? If we find in the future that we need this information elsewhere we decide then what to do.

Reply:
I think that rather sooner than later another some other class apart from EnumerableNestedLoopJoin will need this method (starting with EnumerableCorrelate if/when we deprecate CorrelateJoinType). But I'm not against moving this method into EnumerableNestedLoopJoin for the moment.